stellar-contract-env-host/src/host.rs

```diff
@@ -867,10 +870,54 @@ impl CheckedEnv for Host {
     }
 
     fn compute_hash_sha256(&self, x: Object) -> Result<Object, HostError> {
-        todo!()
+        let binary = self.visit_obj(x, |bin: &Vec<u8>| {
+            Ok(bin.clone()) //TODO:Why is this neccesary
+        })?;
+
+        let mut hasher = Sha256::new();
+        hasher.update(binary);
+        let raw_hash = hasher.finalize();
+
+        let hash: Vec<u8> = raw_hash[0..32]
+            .try_into()
+            .expect("slice with incorrect length");
+        Ok(self.add_host_object(hash)?.into())
```

Comment:
The range 0..32 is going to panic if the hash is less than 32bytes long, but not if it is longer. I don't these few lines do what you need. The try_into() is always going to succeed because there is also an into() defined for this conversion, since byte slices are always convertible into `Vec<u8>`.

If you want to check the length there's a few things you could do:

1. Simply check the length manually.
2. Convert the byte slice into a fixed-size array, and the try_into() should check length.

I think you want to get this into a `Hash` object? So you could convert to a fixed length array and put that into the hash. Something like:
```suggestion
        let hash: [u8; 32] = raw_hash
            .try_into()
            .expect("slice with incorrect length");
        Ok(self.add_host_object(Hash(hash))?.into())
```

It'd be nice if the stellar-xdr crate made this simpler and allowed you to try_into directly from a slice to a Hash, and I'll look at adding that. Hopefully it'd end up something like:
```suggestion
        let hash: Hash = raw_hash[..]
            .try_into()
            .expect("slice with incorrect length");
        Ok(self.add_host_object(hash)?.into())
```

Reply:
I changed how all of this works